### PR TITLE
Define wrappers for /v1/integration APIs

### PR DESCRIPTION
**Pull request by alavers**
Also introduced a DisabledApi. If an incorrect scope is provided, any calls made to the integration APIs will fail immediately, with a friendlier error message.

Also changed webhook APIs to throw immediately when params fail validation, which means this branch will require a major version bump.

### CODE REVIEW

File: src/api/integrations.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    }),
+
+    /**
+     * Fetch the integrations currently configured
+     * @memberof IntegrationsApi.prototype
+     * @method list
+     * @return {APIResponse}
+     */
+    list: smoochMethod({
+        params: [],
```
Comment (mspensieri) [inline src/api/integrations.js]: list integrations can accept a types param to filter by type. Or is this just for path params?

Comment (alavers) [inline src/api/integrations.js]: Params is all params :) I just forgot about `?type=`. Adding...

Comment (mspensieri) [inline src/api/integrations.js]: types with `s`. `?types=messenger,line,viber`

**Comment by alavers** [comment]
^^^

> which means this branch will require a major version bump.

**Comment by lemieux** [comment]
yeah I was mostly replying to @mspensieri ;)

**Comment by alavers** [comment]
OIC!

File: src/api/integrations.js (hunk 1 above)
Comment (alavers) [inline src/api/integrations.js]: Incoming docs bug fix

screen shot 2016-12-22 at 3 28 43 pm

**Pull request merged**